ethereum_history_api/circuits/lib/src/misc/arrays_test.nr

```diff
@@ -51,3 +51,21 @@ fn test_sub_array_equals_index_out_of_bound() -> bool {
     let b: [Field; 5] = [1, 2, 3, 4, 5];
     sub_array_equals(a, b, 3)
 }
+
+#[test]
+fn test_array_equals() {
+    let a: [Field; 0] = [];
+    let b: [Field; 3] = [1, 2, 3];
+    assert(array_equals(a, a));
+    assert(array_equals(b, b));
+}
+
+#[test(should_fail)]
```

Comment:
It should be: assert(array_equals(a,b) | array_equals(c,d)) so that we check that both array_equals fail, not only one.